go-controller/cmd/ovn-k8s-cni-overlay/ovn-k8s-cni-overlay.go

```diff
@@ -151,6 +61,27 @@ func cmdAdd(args *skel.CmdArgs) error {
 	}
 	kubecli := &kube.Kube{KClient: clientset}
 
+	// TODO: Remove the following once the Kubernetes will get
+	// proper Windows CNI support.
+	// NOTE Windows ONLY: there are some cases where we want to return here
+	// and not to continue
+	stop := app.InitialPlatformCheck(args)
+	if stop {
```

Comment:
It is platform specific. There are some cases where kubelet will call CNI ADD when the pod it's getting deleted. This will cause it to hang since it will not be able to fetch the pod annotations and will time out. I'm not sure if just the error would be enough since we don't want to return any error in this case.
But your idea sounds good, gonna move it inside InitialPlatformCheck.